Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,39 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def missingpostiveint(array)
+    array = array.sort 
+    expected = array[0]
+    
+        
+ 
+

```

Comment:
why so much space? you just need one empty line if you want to split this up

Reply:
I'm sorry, i didnt notice that space. will fix it